stripe/src/main/java/com/stripe/android/paymentsheet/DefaultPaymentSheetFlowController.kt

```diff
@@ -121,8 +125,24 @@ internal class DefaultPaymentSheetFlowController internal constructor(
         callback: ApiResultCallback<PaymentIntentResult>
     ) {
         if (data != null && paymentController.shouldHandlePaymentResult(requestCode, data)) {
-            // TODO(mshafrir-stripe): fire payment result analytics
-            paymentController.handlePaymentResult(data, callback)
+            paymentController.handlePaymentResult(
+                data,
+                object : ApiResultCallback<PaymentIntentResult> {
+                    override fun onSuccess(result: PaymentIntentResult) {
+                        if (result.outcome == StripeIntentResult.Outcome.SUCCEEDED) {
+                            eventReporter.onPaymentSuccess(paymentSelection)
+                        } else {
+                            eventReporter.onPaymentFailure(paymentSelection)
+                        }
+                        callback.onSuccess(result)
```

Comment:
It looks like this is called even if the outcome isn't "SUCCEEDED". Is this right?

Reply:
This is a bit of possibly confusing/unfortunate naming. `onSuccess()` refers to the API call being successful, not the underlying object being in a success state.